src/Maps/components/MapsContainer.ts

```diff
@@ -0,0 +1,242 @@
+        this.resetSubscriptions(nextProps.mxObject);
+        const validationMessage = validateLocationProps(nextProps);
+        if (nextProps && nextProps.mxObject) {
+            if (validationMessage) {
+                this.setState({ alertMessage: validationMessage });
+            } else {
+                this.fetchData(nextProps.mxObject);
+            }
+        } else {
+            this.setState({ locations: [], alertMessage: "", isFetchingData: false });
+        }
+    }
+
+    componentWillUnmount() {
+        this.unsubscribeAll();
+    }
+
+    private resetSubscriptions(contextObject?: mendix.lib.MxObject) {
+        this.unsubscribeAll();
+        if (this.props.locations && this.props.locations.length) {
+            if (contextObject) {
+                this.subscriptionHandles.push(window.mx.data.subscribe({
+                    guid: contextObject.getGuid(),
+                    callback: () => this.fetchData(contextObject)
+                }));
+                this.props.locations.forEach(location => {
+                    this.subscriptionHandles.push(window.mx.data.subscribe({
+                        entity: location.locationsEntity as string,
+                        callback: () => this.fetchData(contextObject)
+                    }));
+                    [
+                        location.latitudeAttribute,
+                        location.longitudeAttribute,
+                        location.staticMarkerIcon,
+                        location.markerImageAttribute
+                    ]
+                    .forEach(
+                        (attr): number => this.subscriptionHandles.push(window.mx.data.subscribe({
+                            attr,
+                            callback: () => this.fetchData(contextObject),
+                            guid: contextObject.getGuid()
+                        }))
+                    );
+                });
+            }
+        }
+    }
+
+    private unsubscribeAll() {
+        this.subscriptionHandles.forEach(window.mx.data.unsubscribe);
+        this.subscriptionHandles = [];
+    }
+
+    private fetchData = (contextObject?: mendix.lib.MxObject) => {
+        this.setState({ isFetchingData: true });
+        Promise.all(this.props.locations.map(locationAttr =>
+            this.retrieveData(locationAttr, contextObject)
+        ))
+        .then(locations => {
+            const locs = ([] as Location[]).concat(...locations);
+            if (validLocations(locs)) {
+                this.setState({
+                    locations: locs,
+                    isFetchingData: false,
+                    alertMessage: ""
+                });
+            } else {
+                throw new Error("Invalid Coordinates passed");
+            }
+        })
+        .catch(reason => {
+            this.setState({
+                locations: [],
+                alertMessage: `Failed due to ${reason.message}`,
```

Comment:
Have given feedback on this kind of error before... why u making me look for all of them? :)

Reply:
done